lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
+  return params.reduce((acc, {
+    value, name, type, valueType,
+  }) => ({
+    ...acc,
+    [name]: parsers.resolveParser(type || valueType)(value),
+  }), {});
+}
+
+function handleInput(awsService, actionParams, pluginSettings) {
+  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
+  const client = core.getServiceInstance(awsService, params, settings);
+  return { params, settings, client };
+}
+
+function simpleItemsFilter(autocompleteItems, query) {
+  let result = autocompleteItems;
+  if (query) {
+    const qWords = query.split(/[. ]/g).map((word) => word.toLowerCase());
+    result = result.filter((item) => qWords
+      .every((word) => item.value.toLowerCase().includes(word)))
+      .sort((word1, word2) => word1.value.toLowerCase().indexOf(qWords[0])
```

Comment:
This function does a bit of magic, let's use a meaningful incantation to make casting this spell easier.

Reply:
All the incantations I know are either in Latin or in Elven tongue, so that the mere mortals won't understand them, of course we could make it more meaningful or easier to understand, but where's the fun in that? :)

All jokes aside - as mentioned above, the whole function is to be rewritten.